scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -137,16 +141,49 @@ class RunCoverage(
         println("\nComputed Coverage Ratio is: $computedCoverageRatio")
         println("\nGenerated report at: $reportOutputPath\n")
       }
-    } ?: println("No coverage reports generated.")
-  }
 
-  private fun runCoverageForTarget(testTarget: String): CoverageReport {
-    return runBlocking {
-      CoverageRunner(rootDirectory, scriptBgDispatcher, commandExecutor)
-        .runWithCoverageAsync(testTarget.removeSuffix(".kt"))
-        .await()
+      println("COVERAGE ANALYSIS COMPLETED.")
     }
   }
+
+  private fun runCoverageForTarget(testTarget: String): Deferred<CoverageReport> {
+    return CoverageRunner(rootDirectory, scriptBgDispatcher, commandExecutor)
+      .runWithCoverageAsync(testTarget.removeSuffix(".kt"))
+  }
+}
+
+private fun calculateAggregateCoverageReport(
+  coverageReports: List<CoverageReport>
+): CoverageReport {
+  fun aggregateCoverage(coverages: List<Coverage>): Coverage {
+    return if (coverages.contains(Coverage.FULL)) Coverage.FULL
+    else Coverage.NONE
```

Comment:
```suggestion
    return coverages.find { it == Coverage.FULL } ?: Coverage.NONE
```

Probably can inline this method entirely since it can be done idiomatically in one line.

Reply:
Got it. Updated as suggested.